### ISSUE TITLE
ZIP code in wrong field after editing address

### ISSUE THREAD

**Issue by ngolatka** [netresearch/dhl-module-shipping-m2#54]
**Bug description**
When editing the shipping address for a shipment to a _DHL Postfiliale_ (German post office), the receiver's ZIP code ends up in the wrong request field. It is copied to the receiver city field, which overwrites the city name.

The DHL Versenden API throws this error as a result:
> weak validation error: Der angegebene Ort passt nicht zur Postleitzahl

**Steps to reproduce**
1. Create an order to a _DHL Postfiliale_ using the regular Magento checkout, and make a deliberate mistake (e.g. non-existing post office number) to provoke a failure.
2. Try to create a DHL label, which should fail.
3. Correct the shipping address in the admin panel.
4. Again, try to create a DHL label.
5. Look at the request in the debug.log. The _receiver city_ is filled with the ZIP code.

This problem only occurs when shipping to a _DHL Postfiliale_. It does not occur for _DHL Packstations_, _Paketshops_, or regular postal addresses.

**Expected behavior**
The ZIP code and city name should not get mixed up when editing the shipping address.

**Environment**
 - Module version: 0.10.1

**Comment by ngolatka** [comment]
This is a **known issue in module version 0.10.1**. It will be fixed in future versions (more precisely in the [lib-shipping-mx](https://github.com/netresearch/dhl-lib-shipping-mx) which is used by the DHL Shipping module).

If you can't wait for the next release, apply the following fix manually:
* In the [lib-shipping-mx](https://github.com/netresearch/dhl-lib-shipping-mx), edit the file _Model/ShippingInfo/Receiver/Postfiliale.php_
* Look for this function:
```
public function getCity()
{
    return $this->zip;
}
```

The mistake is rather obvious. Correct it like this:
```
public function getCity()
{
    return $this->city;
}
````

**Comment by moehrenzahn** [comment]
This should be fixed by today's 0.10.2 release which includes a new version of the lib-shipping-mx:  https://github.com/netresearch/dhl-lib-shipping-mx/releases/tag/0.11.1
